### ISSUE TITLE
Statically link libstdc++ when building iree-benchmark-module with tracing enabled

### ISSUE THREAD

**Issue by asaadaldien** [google/iree#3392]
When building with tracing enabled `-DIREE_ENABLE_RUNTIME_TRACING=ON`  it seems like the tracing lib brings `libc++_shared.so` dependency which isn't deployed in android system (see: https://developer.android.com/ndk/guides/cpp-support). 

So instead we need to statically link with libstdc++ (e.g compiling with `clang++ -static-libstdc++`)
 
objdump  iree-benchmark-module release + tracing:
```
Dynamic section at offset 0x1b32d8 contains 30 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so]
 0x0000000000000001 (NEEDED)             Shared library: [libc++_shared.so]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so]
```

objdump  iree-benchmark-module release:
```
Dynamic section at offset 0x1753b8 contains 29 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so]
```

**Comment by benvanik** [comment]
I think this is yet another example of LLVM being annoying - I believe they override things to pull in the dynamic runtime - I'm dealing with this on MSVC right now. I'm leaning more and more toward finding a way to not include LLVM's cmake project in ours given all these stupid global flags that they set :/

**Comment by asaadaldien** [comment]
I think this is the default `clang++` always adds `libc++_shared.so`

**Comment by bjacob** [comment]
Fixed by #3531
